bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gnirs/GNIRSParams.java

```diff
@@ -1014,6 +1014,132 @@ static public AcquisitionMirror getAcquisitionMirror(String name, AcquisitionMir
         }
     }
 
+    /**
+     * Hartmann Masks
+     */
+    public enum HartmannMask implements DisplayableSpType, SequenceableSpType {
+
+        OUT("Out"),
+        LeftMask("Left"),
+        RightMask("Right"),
+        ;
+
+        public static HartmannMask DEFAULT = OUT;
+        private String _displayValue;
+
+        HartmannMask(String displayValue) {
+            _displayValue = displayValue;
+        }
+
+        public String displayValue() {
+            return _displayValue;
+        }
+
+        public String sequenceValue() {
+            return _displayValue;
+        }
+
+        public String toString() {
+            return _displayValue;
+        }
+
+        /**
+         * Return the HartmannMask by name *
+         */
+        static public HartmannMask getHartmannMask(String name) {
+            return getHartmannMask(name, DEFAULT);
+        }
+
+        /**
+         * Return the HartmannMask by index *
+         */
+        static public HartmannMask getHartmannMaskByIndex(int index) {
+            return SpTypeUtil.valueOf(HartmannMask.class, index, DEFAULT);
+        }
+
+        /**
+         * Return the HartmannMask by name giving a value to return upon error *
+         */
+        static public HartmannMask getHartmannMask(String name, HartmannMask nvalue) {
+            return SpTypeUtil.oldValueOf(HartmannMask.class, name, nvalue);
+        }
+    }
+
+    /**
+     * Focus
+     */
+    public enum FocusSuggestion implements DisplayableSpType, SequenceableSpType {
+
+        BEST_FOCUS("best focus"),;
+
+        /**
+         * The default Focus value *
+         */
+        public static final FocusSuggestion DEFAULT = BEST_FOCUS;
+
+        private final String _displayValue;
+
+        FocusSuggestion(String displayValue) {
+            _displayValue = displayValue;
+        }
+
+        public String displayValue() {
+            return _displayValue;
+        }
+
+        public String sequenceValue() {
+            return _displayValue;
+        }
+
+        public String toString() {
+            return _displayValue;
+        }
+
+    }
+
+    public static class Focus extends SuggestibleString {
+        public Focus() {
+            super(FocusSuggestion.class);
+            setStringValue(FocusSuggestion.DEFAULT.displayValue());
+        }
+    }
+
+    public static class FocusEditor extends PropertyEditorSupport {
+        public Object getValue() {
+            Focus f = (Focus) super.getValue();
+            if (f == null) return null;
+
+            Focus res = new Focus();
+            res.setStringValue(f.getStringValue());
+            return res;
```

Comment:
I can see that this is taken from the way that `WavelengthEditor` works.  In fact it pains me to admit I'm originally responsible for this `SuggestibleString` business and the fact that it is so mutable and `null`ish. I wouldn't do it this way these days though.

At any rate, a slight improvement might be to add a `copy` method to `Focus`:

```java
public static final class Focus extends SuggestibleString {
    public Focus(String value) {
        super(FocusSuggestion.class);
        setStringValue(value);
    }

    public Focus() {
        this(FocusSuggestion.DEFAULT.displayValue());
    }

    public Focus copy() {
        return new Focus(getStringValue());        
    }
}
```

and then 

```java
import edu.gemini.shared.util.immutable.*;

public Object getValue() {
    return ImOption.apply((Focus) super.getValue()).map(f -> f.copy()).getOrNull();
}
```

Reply:
Thanks!